### ISSUE TITLE
[TF2] Bumper Cars cause strange behavior once exited.

### ISSUE THREAD

**Issue by StrikerMan780** [ValveSoftware/Source-1-Games#2618]
Once you have the Halloween Bumper Car condition removed, the game goes into a state where you see and hear your weapons shoot twice. It seems you see both the predicted, and the server-side version of everything.

**Comment by ICantShoot** [comment]
I can confirm this bug. I experienced this on last summer when I was making a halloween map with bumper cars. https://steamcommunity.com/sharedfiles/filedetails/?id=1142333364 

Only way to get rid of it is to restart the game as a player.

**Comment by StrikerMan780** [comment]
Glad I'm not the only one. I noticed that the bug can also not crop up until a map change, which really screws things up.

**Comment by StrikerMan780** [comment]
I really hope this gets fixed very soon. This is brutal.

**Comment by Nephyrin** [comment]
Bumper cars disable prediction when entered, so this is very plausible.

Does running something along the lines of `record foo; stop` to force a full client update (or `cl_fullupdate` with cheats enabled) resolve this issue when it occurs?

**Comment by ICantShoot** [comment]
This bug doesn't appear on client servers, only dedicated ones. I'm sure record something;stop doesn't fix it. I recall trying it before many times. I always had to restart my own game to get rid of it. Its most visible with rocket launcher but any shooting gun will do.

**Comment by StrikerMan780** [comment]
Yeah, can confirm that it only seems to happen on dedicated servers, but it screws up prediction for all clients that were connected to the server and had transitioned to another map after being in a bumpercar, forcing them to restart TF2.

**Comment by StrikerMan780** [comment]
I really hope a fix for this is in the next update, because this borderline renders my server unplayable.

**Comment by StrikerMan780** [comment]
Two months, and still no fix. What's even going on?

**Comment by ICantShoot** [comment]
So i'm returning this because i got the bumper cars working in the "valve way" without the touch trigger itself. I used my map Gravestone as a base for this research while updating it. https://steamcommunity.com/sharedfiles/filedetails/?id=1142333364

 If i init the bumpercars through same way as doomsday does, the next map doesnt seem to show any lags anymore on firing as they did with triggers for the players.

So basically, it looks like if community moves into making them work properly without the trigger, this bug is fixed. I have to test more today, so i'll report back if i find something different. If you dont hear from me for the next few days, theres nothing for me to add into this discussion. Thanks.

**Comment by ICantShoot** [comment]
I did some further testing on a live server with 24 players and the double shot effect returned after 1 round ended with bumper cars. It seems like the weapon shot sounds are playing twice few milliseconds away from eachothers. But, when map ended, the sound returned to normal on map change that didnt happen before.

If anyone wants to look deeper, heres the map file: 
http://server2.pelipurkki.fi/~dustin/tf2/2018/pl_rumble_event_rc19.bsp 
Heres the effect on demo:
http://server2.pelipurkki.fi/~dustin/tf2/2018/gravestone_double.dem

**Comment by StrikerMan780** [comment]
The double-shot is basically because the client sees what it predicts, followed by what actually happened on the server. You should only be able to see the predicted result, and also, you should have lag compensation, which seems to be absent when the bug hits.

**Comment by Nephyrin** [comment]
If you are able to replicate this bug, try running `setinfo cl_predict 1` in the console whilst dead (it will not work while spawned).

The bug appears to be related to entering a cart within two seconds of respawning, then exiting the cart after that window (or leaving the server).

**Comment by ICantShoot** [comment]
I tested the issue, managed to repro the bug and tried setinfo cl_predict 1 - it seems to fix the whole issue.

**Comment by StrikerMan780** [comment]
Will test with the new update.

**Comment by StrikerMan780** [comment]
Still busted in the latest update.

**Comment by Nephyrin** [comment]
@StrikerMan780 Can you detail the steps you're using to reproduce this, and the case of bad prediction you're seeing?  The common case with client/server prediction getting out of sync was fixed by the latest patch, but nothing else in the engine toggles prediction regularly so there may be other setups we missed.

**Comment by rowedahelicon** [comment]
Hi I don't mean to necro this but I found a solution maybe? I able to reproduce this by simple giving and removing cond 82, the halloween kart. But I can only replicate this on my servers that had **sv_client_predict** set to **1** instead of the default **-1**. Setting that to -1 stops the bug from happening as far as I can see.

I have provided a visual demonstration here : https://youtu.be/Cll8ItIQY14
